src/assisted_test_infra/test_infra/controllers/node_controllers/tf_controller.py

```diff
@@ -61,7 +68,9 @@ def list_nodes(self) -> List[Node]:
         return list(map(self._tf_vm_to_node, tf_vms))
 
     def _get_vm(self, node_name: str) -> Dict[str, Any]:
```

Comment:
```suggestion
    def _get_vm(self, node_name: str) -> Optional[Dict[str, Any]]:
```
as it can return `None`